packages/grid/x-data-grid/src/components/base/GridFooterPlaceholder.tsx

```diff
@@ -8,5 +8,7 @@ export function GridFooterPlaceholder() {
     return null;
   }
 
-  return <rootProps.slots.footer {...rootProps.slotProps?.footer} />;
+  return (
+    <rootProps.slots.footer {...(rootProps.slotProps?.footer as any) /* FIXME: typing error */} />
+  );
```

Comment:
I fixed the straightforward typing errors but left `FIXME` comments for the ones that I couldn't find an easy answer to.